src/Akka.Persistence.Azure/Util/NameValidator.cs

```diff
@@ -0,0 +1,192 @@
+﻿// -----------------------------------------------------------------------------------------
+// <copyright file="NameValidator.cs" company="Microsoft">
+//    Copyright 2013 Microsoft Corporation
+// 
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//      http://www.apache.org/licenses/LICENSE-2.0
+// 
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+// </copyright>
+// ----------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Akka.Persistence.Azure.Util
+{
+    /// <summary>
+    /// Provides helpers to validate resource names across the Microsoft Azure Storage Services.
+    /// </summary>
+    public static class NameValidator
```

Comment:
The latest Blob SDK doesn't have a name validator, so swiping it from the old SDK code

Reply:
Good idea